internal/pkg/client/interface.go

```diff
@@ -31,20 +31,20 @@ type Reader interface {
 	// List retrieves list of objects for a given namespace and list options. On a
 	// successful call, Items field in the list will be populated with the
 	// result returned from the server.
-	List(ctx context.Context, list runtime.Object, namespace string, options metav1.ListOptions) error
+	List(ctx context.Context, list runtime.Object, namespace string, options *metav1.ListOptions) error
 }
 
 // Writer knows how to create, delete, and update Kubernetes objects.
 type Writer interface {
 	// Create saves the object obj in the Kubernetes cluster.
-	Create(ctx context.Context, obj runtime.Object, options metav1.CreateOptions) error
+	Create(ctx context.Context, obj runtime.Object, options *metav1.CreateOptions) error
 
 	// Delete deletes the given obj from Kubernetes cluster.
 	Delete(ctx context.Context, obj runtime.Object, options *metav1.DeleteOptions) error
 
 	// Update updates the given obj in the Kubernetes cluster. obj must be a
 	// struct pointer so that obj can be updated with the content returned by the Server.
-	Update(ctx context.Context, obj runtime.Object, options metav1.UpdateOptions) error
+	Update(ctx context.Context, obj runtime.Object, options *metav1.UpdateOptions) error
```

Comment:
Can you also add `Apply` to the interface?

Reply:
done